bin/dune

```diff
@@ -0,0 +1,208 @@
+(melange.emit
+ (target exe)
+ (libraries shared)
+ (preprocess
+  (pps melange.ppx))
+ (module_systems
+  (es6 mjs))
+ (runtime_deps bin.mjs node-loader.mjs))
+
+(rule
+ (action
+  (with-stdout-to
+   node_modules.sexp
+   ; can be used to generate files with exe/node_modules/melange,
+   ; exe/node_modules/melange.belt, and exe/node_modules/melange.js
+   (system "echo '(' exe/node_modules/rescript-ssg.shared/* ')'"))))
+
+(install
+ (section bin)
+ (files
+  ; can't include and generate sexp "on the fly" due to "internal dependency cycle" errors in dune
+  (exe/node_modules/melange/arg.mjs as node_modules/melange/arg.mjs)
```

Comment:
This part is scary, I am concerned about maintainability of this chunk.
Do you think it will be fixed on the Dune side some day?

Reply:
I agree. I spent some time trying to improve it, and found a way where we use a bash script to generate the whole `install` stanza with all the files that need to be copied. It works fine with `dune clean`, plays well with other build targets etc.

The only caveats are:
- we will have to update the script whenever new folders need to be copied (shouldn't happen too often)
- whenever the contents of a folder change, we need to run `dune build` twice. The first time it will fail because `include.sexp` still doesn't have the latest contents, but the second time it will work fine.

Please check https://github.com/denis-ok/rescript-ssg/pull/60/commits/9f935cbe1a29b01851f53500fbab5890e002246e and lmk if it looks better now.